bin/instrumentation_generator

```diff
@@ -2,4 +2,14 @@
 
 require_relative '../.instrumentation_generator/instrumentation_generator'
 
+puts <<-EOF
+
+🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨
+
+ WARNING: Your gem will *NOT* be tested until you add it to the CI workflows in `.github/workflows/ci.yml`!!
```

Comment:
follow up PR, maybe this script can edit the action and add the new instrumentation to the list?

Reply:
Yes, I think that's best in a follow-up. As this PR stands, it's not very easy to know _where_ to put it into the workflow file.

In the future, I think a nice evolution might be:
- One YAML file (modified by `instrumentation_generator`) that is the source of truth not only for gem releases, but also for CI in general.
- That YAML file could have things like `sidecar_services: false` or `appraisals: true` or `jruby: false` that inform how things are to be built.
- By consuming that YAML file, we could then dynamically generate the build matrices and do more impressive things. At minimum, it could help us reduce some of the indirection and hacks this PR introduces around skipping things for jruby, and it also could present a way out of the "we can't cache gems for `Appraisals`" problem too.

But ... in the interests of "stop iterating on this and ship at least a modest improvement," I decided to wait. 😆 

cc @plantfansam, with whom I've been chatting about this stuff ☝️

Reply:
> But ... in the interests of "stop iterating on this and ship at least a modest improvement," I decided to wait. 😆

💯

Reply:
Yeah, I think it's great to merge in the instrumentation generator change as-is and then do a follow-on that can update the workflow file as necessary. I'm a little bit wary of the fanciness of deriving things from a single YAML file (very tough to trace a code path that ends in a YAML file), but will reserve judgement until we try 😇

Reply:
True. It could be quite difficult if we're too clever.